### PR TITLE
Implement TableViewSelection Monitor and Tests

### PR DESCRIPTION
**Pull request by rosemaina**
#### What is the purpose of this pull request?
* This adds the TableViewSelection Monitor.

#### Background
#20

#### Checklist

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] I have read the **CONTRIBUTING** guidelines.
- [x] My code follows the coding style of this project.
- [x] My change requires a change to the documentation.
- [x] I have documented all my changes.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.

### CODE REVIEW

File: Sources/Core/UIKit/Other/TableViewSelectionMonitor.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+#if os(iOS) || os(tvOS)
+
+import UIKit
+
+///
+/// A `TableViewSectionMonitor` instance monitors a tableview for changes to
+/// its row selection.
+///
+
```
Comment (eBardX) [inline Sources/Core/UIKit/Other/TableViewSelectionMonitor.swift]: No blank line here please.

File: Tests/UIKit/Other/TableViewSelectionMonitorTests.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@testable import XestiMonitors
+
+internal class TableViewSelectionMonitorTests: XCTestCase {
+    let notificationCenter = MockNotificationCenter()
+    let tableView = UITableView()
+
+    override func setUp() {
+        super.setUp()
+
+        NotificationCenterInjector.inject = {
```
Comment (eBardX) [inline Tests/UIKit/Other/TableViewSelectionMonitorTests.swift]: All on one line please.

File: XestiMonitors.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -123,6 +123,7 @@
 		BDD79C39EC160D102A6C598C591B866C /* DocumentStateMonitor.swift in Sources */ = {isa = PBXBuildFile; fileRef = D4753DA906CAAC43AA95E1CF6AC622AA /* DocumentStateMonitor.swift */; };
 		CF7869DFFFA2FF0445DC4DA6B6D33566 /* DocumentStateMonitor.swift in Sources */ = {isa = PBXBuildFile; fileRef = D4753DA906CAAC43AA95E1CF6AC622AA /* DocumentStateMonitor.swift */; };
 		CFB7376E05C361DEEB6F1F24D5880AD2 /* DocumentStateMonitorTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8650D5F1A9FF403E7779526361A70B01 /* DocumentStateMonitorTests.swift */; };
+		B9E9389C82EFDCDD50CA99CE41B17602 /* Examples in Resources */ = {isa = PBXBuildFile; fileRef = 5ACFE6A7C6A7A80209B33B4555252CDE /* Examples */; };
```
Comment (eBardX) [inline XestiMonitors.xcodeproj/project.pbxproj]: It looks like you (accidentally?) added the `Examples` folder to this project. That does not belong in here.

File: XestiMonitors.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -287,6 +288,10 @@
 		94C9A29513E3A2B1D9B2149AC424E204 /* PedometerMonitor.swift in Sources */ = {isa = PBXBuildFile; fileRef = 65DABE6A22CA528C367957098EE415FA /* PedometerMonitor.swift */; };
 		DA42FE2795736B1FAD220D023C2D6E15 /* PedometerMonitor.swift in Sources */ = {isa = PBXBuildFile; fileRef = 65DABE6A22CA528C367957098EE415FA /* PedometerMonitor.swift */; };
 		1656F7A2C0D8E6D80B9ADF10CD463919 /* PedometerMonitorTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 29671A0F6146361CA1CBCB2000C8D368 /* PedometerMonitorTests.swift */; };
+		E0A04370E6AD3F325FB73ACBC4E2112E /* Pods_XestiMonitors_iOS.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 01D187F8D6CE6FC4501E037CEAAB548C /* Pods_XestiMonitors_iOS.framework */; };
```
Comment (eBardX) [inline XestiMonitors.xcodeproj/project.pbxproj]: Ditto for the `Pods` folder. Please remove it.

File: XestiMonitors.xcodeproj/project.pbxproj (hunk 3 above)
Comment (rosemaina) [reply XestiMonitors.xcodeproj/project.pbxproj]: I have removed it.

**Pull request merged**